glide/blog.md

```diff
@@ -0,0 +1,147 @@
+
+---
+
+<br>
+
+I believe browsers should be hackable, just like your editor.
+
+```typescript
+glide.keymaps.set("normal", "gC", async () => {
+  // extract the owner and repo from a url like 'https://github.com/glide-browser/glide'
+  const [owner, repo] = glide.ctx.url.pathname.split("/").slice(1, 3);
+  if (!owner || !repo) throw new Error("current URL is not a github repo");
+
+  // * clone the current github repo to ~/github.com/$owner/$repo
+  // * start kitty with neovim open at the cloned repo
+  const repo_path = glide.path.join(glide.path.home_dir, "github.com", owner, repo);
+  await glide.process.execute("gh", [ "repo", "clone", glide.ctx.url, repo_path]);
+  await glide.process.execute("kitty", ["-d", repo_path, "nvim"], { cwd: repo_path });
+}, { description: "open the GitHub repo in the focused tab in Neovim" });
+```
+
+> Example Glide keymapping to clone the GitHub repo in the current tab and open it in [Kitty](https://sw.kovidgoyal.net/kitty/) + [Neovim](https://neovim.io/).
+
+This snippet captures exactly the kind of functionality I personally want from a browser; extreme customisability so that I can make my daily workflows a joy.
+
+### What's wrong with the status quo?
+
+Most browsers only allow you to customise shortcuts for certain predefined actions, and do not allow you to define your own entirely custom keymappings without going through a web extension; and even then support for writing your own code is very limited.
+
+Existing web extensions[^2] do offer similar functionality to Glide but are artificially restricted in scope by security constraints[^3] imposed on all web extensions.
+
+For example, extensions are completely disabled on certain URIs, e.g. [addons.mozilla.org](https://addons.mozilla.org), the json file viewer, etc. Which means that relying on extension defined mappings on those pages will not work.
+
+There are also *many* useful APIs that are not provided to web extensions, such as arbitrary file writes, spawning processes, and setting prefs.
+
+### How is Glide different?
+
+Glide holistically solves these usability issues by introducing [modes](#modes), and a TypeScript [config](https://glide-browser.app/config) that lets you do _anything_[^4] you'd like to.
+
+You can define custom [key mappings](https://glide-browser.app/keys), access the [web extensions API](https://glide-browser.app/extensions), spawn arbitrary [processes](https://glide-browser.app/api#glide.process), define [macros](https://glide-browser.app/api#glide.keys) and more.
+
+Glide can support APIs and functionality that will never be supported in web extensions as the security model is flipped. You - the end user - are responsible for writing the config code, so there is no reason to restrict what you can do.
+
+Here's a small example that adds g + c as a key mapping to switch to the calendar tab:
```

Comment:
```suggestion
Here's a small example that adds <key>g</key>+<key>c</key> as a key mapping to switch to the calendar tab:
```

if you can style this it's kinda nice... not sure how bear blog renders this

Reply:
just gets stripped :(

Reply:
wait wdym I can't even copy that symbol lol

Reply:
oh lmfao i did not expect that to be the way that rendered